src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs

```diff
@@ -6501,6 +6501,91 @@ static IEnumerable<int> F()
                 null);
         }
 
+        [Fact]
+        public void CSharp7SwitchStatement()
+        {
+            var src1 = @"
+class C
+{
+    static void F(object o)
+    {
+        switch (o)
+        {
+            case int i:
+                break;
+        }
+        System.Console.WriteLine(1);
+    }
+}
+";
+            var src2 = @"
+class C
+{
+    static void F(object o)
+    {
+        switch (o)
+        {
+            case int i:
+                break;
+        }
+        System.Console.WriteLine(2);
+    }
+}
+";
+            var edits = GetTopEdits(src1, src2);
+
+            CSharpEditAndContinueTestHelpers.Instance.VerifySemantics(
+                edits,
+                ActiveStatementsDescription.Empty,
+                expectedDiagnostics: new[]
+                {
+                    Diagnostic(RudeEditKind.UpdateAroundActiveStatement, "switch (o)", CSharpFeaturesResources.v7_switch)
+                });
+        }
+
+        [Fact]
+        public void AddCSharp7SwitchStatement()
+        {
+            var src1 = @"
+class C
+{
+    static void F(object o)
+    {
+    }
+}
+";
+            var src2 = @"
+class C
+{
+    static void F(object o)
+    {
+        switch (o)
+        {
+            case string s:
+                break;
+        }
+        switch (o)
+        {
+            case int i:
+                break;
+        }
+    }
+}
+";
+            var edits = GetTopEdits(src1, src2);
+
+            CSharpEditAndContinueTestHelpers.Instance.VerifySemantics(
+                edits,
+                ActiveStatementsDescription.Empty,
+                additionalOldSources: null,
+                additionalNewSources: null,
+                expectedSemanticEdits: null,
+                expectedDiagnostics: new[]
+                {
+                    Diagnostic(RudeEditKind.UpdateAroundActiveStatement, "switch (o)", CSharpFeaturesResources.v7_switch)
+                });
+        }
+
```

Comment:
Can you please add a test that uses a "new" switch with a V6-compatible syntax? e.g. switch on object and have "case 1" and "case string.Empty"?